entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -346,6 +350,70 @@ public void bulkUpdate(
     }
   }
 
+  @Override
+  public void bulkUpdateEntityArrayAttribute(
+      BulkEntityArrayAttributeUpdateRequest request,
+      StreamObserver<ResultSetChunk> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tenantId = requestContext.getTenantId().orElse(null);
+    if (isNull(tenantId)) {
+      responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
+      return;
+    }
+    try {
+      Set<Key> keys =
+          request.getEntityIdsList().stream()
+              .map(entityId -> new SingleValueKey(tenantId, entityId))
+              .collect(Collectors.toSet());
+
+      String attributeId = request.getAttribute().getColumnName();
+
+      String subDocPath =
+          entityAttributeMapping
+              .getDocStorePathByAttributeId(requestContext, attributeId)
+              .orElseThrow(
+                  () -> new IllegalArgumentException("Unknown attribute FQN " + attributeId));
```

Comment:
nit (FQN in the context of attributes is deprecated, we use attribute IDs now):
```suggestion
                  () -> new IllegalArgumentException("Unknown attribute " + attributeId));
```